sql/api/src/main/scala/org/apache/spark/sql/internal/types/AbstractStringType.scala

```diff
@@ -21,25 +21,35 @@ import org.apache.spark.sql.internal.SqlApiConf
 import org.apache.spark.sql.types.{AbstractDataType, DataType, StringType}
 
 /**
- * AbstractStringType is an abstract class for StringType with collation support. As every type of
- * collation can support trim specifier this class is parametrized with it.
+ * AbstractStringType is an abstract class for StringType with collation support.
  */
-abstract class AbstractStringType(private[sql] val supportsTrimCollation: Boolean = false)
+abstract class AbstractStringType(supportsTrimCollation: Boolean = false)
```

Comment:
why did you remove `private[sql] val`? let's keep it private unless there's a need

Reply:
Why does it need to private? Looks like other classes inheriting from AbstractDataType like AbstractMapType don't have their members private either.

Reply:
can we keep it private to SQL in all of them? no need to expose stuff unless there's a need to expose stuff

Reply:
`sql/internal` is already a private package so we won;t need to mark it as `private[sql]`.   We should add a `package.scala` and document so, see https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/package.scala

Reply:
See also https://issues.apache.org/jira/browse/SPARK-16964 and https://issues.apache.org/jira/browse/SPARK-16813